### DISCUSSION TITLE
Sensible terminal commands not working when bound

### DISCUSSION THREAD

**Discussion by ghbrown** [i3/i3#5044] [Q&A]
I have `$TERMINAL` set in my `.profile`, but both rofi-sensible-terminal and i3-sensible-terminal fail to find it when bound in i3.
For example,
```
$ i3-sensible-terminal
```
opens termite (which is what I set in `${TERMINAL}`, good), but
```
bindsym o exec i3-sensible-terminal
```
opens uxterm.

This also extends to rofi-sensible-terminal, and impacts how rofi opens terminal applications. For example,
```
$ rofi -modi drun -show drun -config <config_path>
```
runs a rofi that opens terminal applications in `${TERMINAL}` (good), while
```
bindsym o exec rofi -modi drun -show drun -config <config_path>
```
runs a rofi that opens terminal applications in uxterm.

Nevertheless, it seems setting `${TERMINAL}` "should work" to ensure that rofi opens terminal applications in `${TERMINAL}` according to the maintainer of the i3 version of my distro (EndeavourOS) [here](https://forum.endeavouros.com/t/set-default-terminal-for-applications-launched-from-rofi-menu/26452). Others have also had success, but for some reason it's not working for me, and I feel that the issue is more i3 than rofi, so I've come here hoping for some wisdom.

Please let me know if I can provide any more relevant information.

**Comment by vincentbernat** [comment]
Once you are in uxterm, what's the value of `echo $TERMINAL`?

**Reply by ghbrown** [reply]
I bound to `i3-sensible-terminal`, and the value of `${TERMINAL}` was termite (as I set in .profile).

**Reply by vincentbernat** [reply]
Your profile is read only if you have a shell on the path to running i3. How do you run i3?

**Reply by ghbrown** [reply]
I'm using the i3 version of EndeavourOS, so I didn't set this up myself. However, I looked through the [repo that sets up i3 for the distro](https://github.com/endeavouros-team/endeavouros-i3wm-setup) and didn't see anything obviously related to xinit (which is all I really know to look for).

The i3_install script (which I think only runs once?) is
```bash
#!/usr/bin/env bash 
git clone https://github.com/endeavouros-team/endeavouros-i3wm-setup.git
cd endeavouros-i3wm-setup
cp -R .config ~/                                            
chmod -R +x ~/.config/i3/scripts
cp .nanorc ~/
dbus-launch dconf load / < ~/xed.dconf
sed -i 's|\(exec --no-startup-id ~/set_once.sh\)|# \1|' ~/.config/i3/config
wget https://raw.githubusercontent.com/endeavouros-team/EndeavourOS-packages-lists/master/i3
sudo pacman -S --needed - < i3
cd ..
rm -rf endeavouros-i3wm-setup
```

If you could let me know where else I can look to answer your question I'd appreciate it.

**Reply by ghbrown** [reply]
Looks like it's started in `.xinitrc` via
```bash
get_session(){
        local dbus_args=(--sh-syntax --exit-with-session)
        case $1 in
                awesome) dbus_args+=(awesome) ;;
                bspwm) dbus_args+=(bspwm-session) ;;
                budgie) dbus_args+=(budgie-desktop) ;;
                cinnamon) dbus_args+=(cinnamon-session) ;;
                deepin) dbus_args+=(startdde) ;;
                enlightenment) dbus_args+=(enlightenment_start) ;;
                fluxbox) dbus_args+=(startfluxbox) ;;
                gnome) dbus_args+=(gnome-session) ;;
                i3|i3wm) dbus_args+=(i3 --shmlog-size 0) ;;
                jwm) dbus_args+=(jwm) ;;
                kde) dbus_args+=(startplasma-x11) ;;
                lxde) dbus_args+=(startlxde) ;;
                lxqt) dbus_args+=(lxqt-session) ;;
                mate) dbus_args+=(mate-session) ;;
                xfce) dbus_args+=(xfce4-session) ;;
                openbox) dbus_args+=(openbox-session) ;;
                *) dbus_args+=($DEFAULT_SESSION) ;;
        esac

        echo "dbus-launch ${dbus_args[*]}"
}

exec $(get_session "$1")
```

**Reply by ghbrown** [reply]
Any updates on what might be happening here?
I recognize this is not a widely encountered issue, but I'd love to get closure one way or another. Thanks!

**Comment by orestisfl** [comment]
`startx` does not source `~/.profile`, it must be sourced by your shell. Bash does not always source that file: https://superuser.com/a/121385

Since `$TERMINAL` is an X-specific variable, I would `export` it in `~/.xinitrc`.
